src/AutoRest.CSharp/AutoRest/Plugins/Configuration.cs

```diff
@@ -34,10 +36,13 @@ public Configuration(string outputFolder, string? ns, string? name, string[] sha
         public bool ModelNamespace { get; }
         public bool HeadAsBoolean { get; }
         public bool SkipCSProjPackageReference { get; }
+        public string[] CredentialTypes { get; }
+        public string[]? CredentialScopes { get; }
```

Comment:
Don't make the array nullable, default it to empty instead. Would avoid having to check for null at all callsites.